libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -515,12 +515,14 @@ class MapboxNavigation(
      * Use [RoutesObserver] and [MapboxNavigation.registerRoutesObserver] to observe whenever the routes list reference managed by the SDK changes, regardless of a source.
      *
      * @param routes a list of [DirectionsRoute]s
+     * @param legIndex starting leg to follow
      * @see [requestRoutes]
      */
-    fun setRoutes(routes: List<DirectionsRoute>) {
+    @JvmOverloads
+    fun setRoutes(routes: List<DirectionsRoute>, legIndex: Int = 0) {
```

Comment:
```suggestion
    fun setRoutes(routes: List<DirectionsRoute>, initialLegIndex: Int = 0) {
```
can we replace everywhere `legIndex` with `initialLegIndex ` to make it more clear?